### PR TITLE
Update signing component to make formatting easier and more visible

### PR DESCRIPTION
**Pull request by TommyHuang-dev**
Also fixes signing component routing away even if there was an error

### CODE REVIEW

File: src/app/pages/signing/signing.component.ts [hunk 1]
```diff
@@ -62,7 +63,18 @@ export class SigningComponent implements OnInit {
...
-      this.router.navigate(['/msg-sent'], { queryParams: { name: this.name } }))
-      .catch(error => alert(error));
+      void (0))
+      .catch(error => {
+          alert(error);
+          valid = false;
+        }
+      );
+    // set timeout waits until after the next event to do stuff
+    // hopefully this works :)
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.ts:73]:
```suggestion
```

File: src/app/pages/signing/signing.component.ts [hunk 2]
```diff
@@ -62,7 +63,18 @@ export class SigningComponent implements OnInit {
...
     this.msgIo.sendMessage(newMsg, this.key).then(() =>
-      this.router.navigate(['/msg-sent'], { queryParams: { name: this.name } }))
-      .catch(error => alert(error));
+      void (0))
+      .catch(error => {
+          alert(error);
+          valid = false;
+        }
+      );
+    // set timeout waits until after the next event to do stuff
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.ts:72]:
```suggestion
    // Wait until after the next event to navigate
```

File: src/app/pages/maintenance/maintenance.component.html [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<div class="container vertical-center montserrat">
+  <div class="row justify-content-center mx-3">
+    <h1 class="maintenance-heading">Our database is currently under maintenance. While you can view your virtrolio as
+      usual, sending messages will be unavailable for the time being</h1>
+    <h2 class="maintenance-subheading">We apologize for the inconvenience</h2>
```
Comment (ricetech) [inline src/app/pages/maintenance/maintenance.component.html:5]: Missing period
```suggestion
    <h2 class="maintenance-subheading">We apologize for the inconvenience.</h2>
```

File: src/app/pages/maintenance/maintenance.component.html [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<div class="container vertical-center montserrat">
+  <div class="row justify-content-center mx-3">
+    <h1 class="maintenance-heading">Our database is currently under maintenance. While you can view your virtrolio as
+      usual, sending messages will be unavailable for the time being</h1>
```
Comment (ricetech) [inline src/app/pages/maintenance/maintenance.component.html:4]:
```suggestion
    <h1 class="maintenance-heading">Our database is currently under maintenance. While you can still
      view <a routerLink="/your-virtrolio">your virtrolio</a> and use the rest of the website,
      sending messages will be unavailable until maintenance is complete.</h1>
```

File: src/app/core/signing.guard.ts [hunk 5]
```diff
@@ -29,55 +30,59 @@ export class SigningGuard implements CanActivate {
...
-    try {
-      const linkStr = window.location.href;
-      SigningGuard.uid = linkStr.match(/uid=([^&]*)/)[1];
-      SigningGuard.key = linkStr.match(/key=([^&]*)/)[1];
-    } catch (error) {
-      // Displaying alert is not necessary since an error is only thrown when one of the query params is missing/invalid = invalid link
-      this.router.navigate([ '/invalid-link' ]).catch(error => AuthService.displayError(error));
-    }
+    if (this.maintenance) {
+      this.router.navigate([ '/maintenance' ]);
```
Comment (ricetech) [inline src/app/core/signing.guard.ts:34]: Missing return statement
```suggestion
      this.router.navigate([ '/maintenance' ]);
      return false;
```

File: src/app/pages/maintenance/maintenance.component.html (hunk 4 above)
Comment (TommyHuang-dev) [reply src/app/pages/maintenance/maintenance.component.html:4]: Added buttons instead

File: src/app/pages/maintenance/maintenance.component.html [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<div class="container vertical-center montserrat">
+    <div class="row justify-content-center m-2 ">
+      <h1 id="maintenance-heading">Our database is currently under maintenance. While you can view your virtrolio as
+        usual, sending messages will be unavailable for the time being.</h1>
```
Comment (ricetech) [inline src/app/pages/maintenance/maintenance.component.html:4]: Add link
```suggestion
      <h1 id="maintenance-heading">Our database is currently under maintenance. While you can view 
        <a routerLink="/your-virtrolio">your virtrolio</a> as usual, sending messages will be unavailable 
        for the time being.</h1>
```

File: src/app/pages/maintenance/maintenance.component.html [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+      <h2 id="maintenance-subheading">We apologize for the inconvenience.</h2>
+    </div>
+    <div class="row justify-content-center mx-3 mt-3">
+      <img src="../../../assets/images/InvalidLink.png" alt="black cloud graphic" id="graphic">
+    </div>
+    <div class="row justify-content-center m-2">
+      <button class="btn-lg btn-primary mt-5 mb-5 mx-3" routerLink="/your-virtrolio">Your virtrolio</button>
+      <button class="btn-lg btn-primary mt-5 mb-5 mx-3" routerLink="/">Back to Home</button>
```
Comment (ricetech) [inline src/app/pages/maintenance/maintenance.component.html:12]: Condense margin selectors
```suggestion
      <button class="btn-lg btn-primary my-5 mx-3" routerLink="/your-virtrolio">Your virtrolio</button>
      <button class="btn-lg btn-primary my-5 mx-3" routerLink="/">Back to Home</button>
```

File: src/app/pages/signing/signing.component.html [hunk 8]
```diff
@@ -104,12 +105,12 @@
 </div>
 
 <!-- Message confirmation modal -->
-<div class="modal fade" id="submit-confirm-modal" tabindex="-1" role="dialog"
-     aria-labelledby="submit-confirm-modal-title" aria-hidden="true">
+<div class="modal fade" id="submit-confirm" tabindex="-1" role="dialog"
+     aria-labelledby="submit-confirm-modal" aria-hidden="true">
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.html:109]:
```suggestion
     aria-labelledby="submit-confirm-title" aria-hidden="true">
```

File: src/app/pages/signing/signing.component.html [hunk 9]
```diff
@@ -104,12 +105,12 @@
...
   <div class="modal-dialog modal-dialog-centered" role="document">
     <div class="modal-content">
       <div class="modal-header">
-        <h5 class="modal-title" id="submit-confirm-modal-title">Are you sure you want to send this?</h5>
+        <h5 class="modal-title" id="submit-confirm-modal">Are you sure you want to send this?</h5>
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.html:113]:
```suggestion
        <h5 class="modal-title" id="submit-confirm-title">Are you sure you want to send this?</h5>
```

File: src/app/pages/signing/signing.component.html [hunk 10]
```diff
@@ -129,12 +130,12 @@ <h5 class="modal-title" id="submit-confirm-modal-title">Are you sure you want to
 </div>
 
 <!-- Information modal when they click on the ? -->
-<div class="modal fade" id="info-modal" tabindex="-1" role="dialog" aria-labelledby="info-modal-title"
+<div class="modal fade" id="info" tabindex="-1" role="dialog" aria-labelledby="info-modal-title"
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.html:133]:
```suggestion
<div class="modal fade" id="info" tabindex="-1" role="dialog" aria-labelledby="info-title"
```

File: src/app/pages/signing/signing.component.html [hunk 11]
```diff
@@ -129,12 +130,12 @@ <h5 class="modal-title" id="submit-confirm-modal-title">Are you sure you want to
...
      aria-hidden="true">
   <div class="modal-dialog modal-dialog-centered" role="document">
     <div class="modal-content">
       <div class="modal-header">
-        <h5 class="modal-title" id="info-modal-title">Craft your perfect message!</h5>
+        <h5 class="modal-title" id="info-modal">Craft your perfect message!</h5>
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.html:138]:
```suggestion
        <h5 class="modal-title" id="info-title">Craft your perfect message!</h5>
```

File: src/app/pages/home/home.component.css [hunk 12]
```diff
@@ -212,7 +212,7 @@ markdown {
 
 .dropdown-menu {
   overflow-y: auto;
-  height: 12em;
+  height: 15em;
```
Comment (ricetech) [inline src/app/pages/home/home.component.css:215]:
```suggestion
  height: 12em;
```

File: src/app/pages/signing/signing.component.ts [hunk 13]
```diff
@@ -61,8 +61,11 @@ export class SigningComponent implements OnInit {
 
     // remove navigation popup
     this.sending = true;
-    this.msgIo.sendMessage(newMsg, this.key).then(() =>
-      this.router.navigate(['/msg-sent'], { queryParams: { name: this.name } }))
-      .catch(error => alert(error));
+    this.msgIo.sendMessage(newMsg, this.key).then(() => {
+      this.router.navigate([ '/msg-sent' ], { queryParams: { name: this.name } });
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.ts:65]: Add catch for navigate
```suggestion
      this.router.navigate([ '/msg-sent' ], { queryParams: { name: this.name } })
        .catch(e => AuthService.displayError(e));;
```

File: src/app/pages/signing/signing.component.ts [hunk 14]
```diff
@@ -61,8 +61,11 @@ export class SigningComponent implements OnInit {
...
+    }).catch(error => {
+        alert(error);
```
Comment (ricetech) [inline src/app/pages/signing/signing.component.ts:67]: Use `AuthService.displayError()`
```suggestion
        AuthService.displayError(error);
```

File: src/app/pages/signing/signing.component.css [hunk 15]
```diff
@@ -93,7 +93,7 @@ markdown {
 
 .dropdown-menu {
   overflow-y: auto;
-  height: 12em;
+  height: 15em;
```
Comment (GuyInFridge) [inline src/app/pages/signing/signing.component.css:96]:
```suggestion
  height: 12em;
```

**Pull request merged**